rfcs/convergence/exposing-popper-options.md

```diff
@@ -0,0 +1,148 @@
+
+- `coverTarget` - allows the positioned element to cover the target
+
+There are also additional options that are not used by any converge component that might be used in the future like:
+
+- `autoSize` - Allows the positioned element to change size based on viewport size
+- `flipBoundary` - Determines what bounds to use to flip the positioning
+
+Issues and bugs can be triggered because certain use cases require more precise configuration of popper. The most likely
+result would be that a popper configuration is exposed for one component but not others.
+
+## Detailed Design or Proposal
+
+### Equal popper configuration
+
+Each converged component that uses popper should allow an equal level of configuration. This ensures that our positioned
+components can behave consistently across the library. Any fixes that require exposing another popper configuration will
+be available to **all** components that use popper.
+
+This proposal does not mean that all popper options must be public, but that if one component exposes a popper configuration,
+then all components should do so equally.
+
+We can achieve this with an agreement that the current `PositioningProps` would be supported by all v9 components, while
+internal popper options can still belong to `PopperOptions` type in the `react-positioning` package.
+
+```tsx
+import { PositioningProps } from '@fluentui/react-positioning';
+
+export interface TooltipProps extends PositioningProps {}
+
+export interface MenuProps extends PositioningProps {}
+
+export interface PopoverProps extends PositioningProps {}
+```
+
+### `positioning` configuration object
+
+The next step to this proposal is to make sure that all components which use a positioned element should allow users
+to configure popper through a `positioning` configuration object:
+
+```tsx
+import { PositioningProps } from '@fluentui/react-positioning';
+
+export interface TooltipProps {
+  positioning?: PositioningProps;
+}
+
+export interface MenuProps {
+  positioning?: PositioningProps;
+}
+
+export interface PopoverProps {
+  positioning?: PositioningProps;
+}
+```
+
+Slots that use popper should use a similar pattern if a top level `positioning` option is not clear enough:
+
+```tsx
+import { PositioningProps } from '@fluentui/react-positioning';
+
+export interface ComponentWithPositionedSlotProps {
+  positionedSlot: React.HTMLAttributes<HTMLElement> & { positioning?: PositioningProps };
+}
+```
+
+### Pros and Cons
+
+#### Pros
+
+- Positioning capability is consistent across the library
+- Fixes to positioned elements through extra configuration can be used across the library
+- Configuring positioned elements is consistent across components and clear for users
+- If a better underlying library than `Popper` exists, we can replace it by implementing the `PositioniningProps`interface
+- Technically a new `positioning`prop is a breaking API change, but can be done in a non-breaking way
```

Comment:
```suggestion
- Technically a new `positioning` prop is a breaking API change, but can be done in a non-breaking way
```

Reply:
Can you please clarify how?

Reply:
Added clarification in the proposal for `placement` prop

Reply:
```suggestion
- Technically a new `placement` prop is a breaking API change, but can be done in a non-breaking way
```